var/spack/repos/builtin/packages/protobuf/package.py

```diff
@@ -53,7 +54,7 @@ class Protobuf(CMakePackage):
 
     depends_on('zlib')
 
-    conflicts('%gcc@:4.6')  # Requires c++11
+    conflicts('%gcc@:4.6', when='@3.6.0:')  # Requires c++11
 
```

Comment:
Why does this only apply to versions >= 3.6 when it previously applied to all versions?

Reply:
[The protobuf release page](https://github.com/protocolbuffers/protobuf/releases) states that c++11 is only required from version 3.6 onward. In particular it says 

> Starting from this release, we now require C++11. For those we cannot yet upgrade to C++11, we will try to keep the 3.5.x branch updated with critical bug fixes only. If you have any concerns about this, please comment on issue https://github.com/protocolbuffers/protobuf/issues/2780.

I have to admit that I didn't bother to check why someone added that conflict in the first place. I have now stumbled across https://github.com/spack/spack/pull/3948 where the conflict was added but it doesn't really make sense to me. If the build really fails because of a gcc version < 4.6 and c++11 is the reason someone should probably inform the protobuf developers about that.

Reply:
> If the build really fails because of a gcc version ...

Can we try this before we lower the restriction? :)

Reply:
> Can we try this before we lower the restriction? :)

I will try it :)

Reply:
Turns out it is not really trivial to do `spack install gcc@4.5.4` maybe someone should rework that package as well :(
I will keep trying :)

Reply:
Yes, I know. Installing older GCC with newer GCC releases turns out to be non-trivial, since usually one would only do it the other way around :see_no_evil:

Reply:
I tested with a `gcc@4.4.7` on a centos6 and it works for all versions of the package except 3.1, 3.2 and as expected 3.6.1 (c++11). I will modify the conflict accordingly soon.

Reply:
I tested with a `gcc@4.4.7` on a centos6 and it works for all versions of the package except 3.1, 3.2 and as expected 3.6.1 (c++11). I will modify the conflict accordingly soon.

Reply:
I tested with a `gcc@4.4.7` on a centos6 and it works for all versions of the package except 3.1, 3.2 and as expected 3.6.1 (c++11). I will modify the conflict accordingly soon.

Reply:
Thanks for testing!